.github/workflows/python_unit_tests.yml

```diff
@@ -0,0 +1,25 @@
+name: Python Unit Tests
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```

Comment:
LGTM --

Question: This seems like a duplicate dependency installation as the make file also handles requirements install?

Also where is matrix.python-version env coming from?

Reply:
and It is duplicate in the sense of the code is duplicated. I can just use a make command to simplify!

Probably don't need the matrix here, but theoretically we could run on multiple python environments
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Reply:
Ahh i see. I'm good with this, thanks for handling this!

#7